src/vs/workbench/api/common/extHostLanguageModels.ts

```diff
@@ -291,54 +291,62 @@ export class ExtHostLanguageModels implements ExtHostLanguageModelsShape {
 		this._onDidChangeProviders.fire(undefined);
 	}
 
+	async getLanguageModelByIdentifier(extension: IExtensionDescription, identifier: string): Promise<vscode.LanguageModelChat | undefined> {
+
+		const data = this._allLanguageModelData.get(identifier);
+		if (!data) {
+			// model gone? is this an error on us?
+			return;
+		}
+
+		// make sure auth information is correct
+		if (this._isUsingAuth(extension.identifier, data.metadata)) {
+			await this._fakeAuthPopulate(data.metadata);
+		}
+
+		let apiObject = data.apiObjects.get(extension.identifier);
+		if (!apiObject) {
+			const that = this;
+			apiObject = {
+				id: data.metadata.id,
```

Comment:
@jrieken this line changed, seems like it was returning a model with the internal ID?

Reply:
Nice catch!